### ISSUE TITLE
Crash on 1.21

### ISSUE THREAD

**Issue by sternschnaube** [Quplet/NoShieldDelay#2]
Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'no-shield-delay'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.15.11.jar:?]
	at net.minecraft.server.Main.main(Main.java:111) [server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [minecraft_server.jar:1.0.1]
Caused by: java.lang.ExceptionInInitializerError
	at qu.noshielddelay.NoShieldDelay.onInitialize(NoShieldDelay.java:20) ~[no-shield-delay-1.0.0.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.11.jar:?]
	... 7 more
Caused by: java.lang.UnsupportedOperationException: StampedConfig does not support valueMap() yet.
	at com.electronwill.nightconfig.core.concurrent.StampedConfig.valueMap(StampedConfig.java:472) ~[com_electronwill_night-config_core-3.7.1-76412f86d3fd480a.jar:?]
	at com.electronwill.nightconfig.core.utils.UnmodifiableConfigWrapper.valueMap(UnmodifiableConfigWrapper.java:29) ~[com_electronwill_night-config_core-3.7.1-76412f86d3fd480a.jar:?]
	at com.electronwill.nightconfig.core.ConfigSpec.isCorrect(ConfigSpec.java:546) ~[com_electronwill_night-config_core-3.7.1-76412f86d3fd480a.jar:?]
	at qu.noshielddelay.config.ModConfig.checkConfig(ModConfig.java:32) ~[no-shield-delay-1.0.0.jar:?]
	at qu.noshielddelay.config.ModConfig.<clinit>(ModConfig.java:19) ~[no-shield-delay-1.0.0.jar:?]
	at qu.noshielddelay.NoShieldDelay.onInitialize(NoShieldDelay.java:20) ~[no-shield-delay-1.0.0.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.11.jar:?]
	... 7 more

**Comment by Quplet** [comment]
I need more information than the beginning of a crash log. I'm unable to recreate this crash myself.

What mods do you have installed with it? Did you change anything in the config file?

**Comment by mpustovoi** [comment]
I have the same issue
[Crash report](https://mclo.gs/OrcRZyp) and [latest.log](https://mclo.gs/0hbwZ6J)

**Comment by Quplet** [comment]
Would you be able to zip and send me all the mods you have installed and the config file generated for my mod?

**Comment by venaminecraft** [comment]
Sorry for the late answer @Quplet.
I hope @mpustovoi helps out, otherwise I can send you my list over / mods and config directory 😄

**Comment by Quplet** [comment]
Either one of you would be great. I would like to replicate the exact environment.

**Comment by mpustovoi** [comment]
> Would you be able to zip and send me all the mods you have installed and the config file generated for my mod?

Honestly, my modpack doesn't work even without your mod. I don't know why. Is says this when I enter the main menu:
![image](https://github.com/Quplet/NoShieldDelay/assets/92390626/1d5e32f2-5498-4529-89c2-b5e0dab15604)
It happened after clicking the `Update all` button in Modrinth App.
The config of your mod is okay I think:
```toml
#This is a Fabric recreation of Revvilo's Responsive Shield's mod. Their mod is exclusively for Forge and I wanted a Fabric one for myself, so here you go.
#Minecraft by default has a hard-coded 5 tick delay between right-clicking with the shield and being able to block.
#Setting RaiseTime to 0 makes shields capable of blocking instantly. Setting it to 5 would be equivalent to vanilla.
Enabled = true
RaiseTime = 0
```

**Comment by mpustovoi** [comment]
In addition, the weight of my `mods` folder is 350 MB. I don't think GitHub will allow the archive to be sent.

**Comment by Quplet** [comment]
If it's an official mod pack, linking that would suffice.

**Comment by mpustovoi** [comment]
Unfortunately, it's my custom modpack.

**Comment by mpustovoi** [comment]
I can send a `.mrpack` which can be automatically installed in Modrinth App. It weighs about 2 Mb.
However, since `.mrpack` isn't supported here, I have archived it.
[Fabric 1.21 broken pack.zip](https://github.com/user-attachments/files/16072552/Fabric.1.21.broken.pack.zip)

**Comment by Quplet** [comment]
I use prism launcher but it should also support it. I'll test and see what I get the chance later tonight. Thank you.

**Comment by mpustovoi** [comment]
Btw, I found the reason for my resource loading problem with my modpack. By the half-exclusion method...
It was caused by the `Creeper Overhaul` mod.
Also, the mods `Cooking for Blockheads` and `Farming for Blockheads` cause problems.

**Comment by KAGUYA-1137** [comment]
**I have the same problem.**
[Crash-report](https://mclo.gs/aN6xOkf)
[latest.log](https://mclo.gs/4nWkzfe)

**Comment by Quplet** [comment]
Ok, i think i found the source of the problem. It has to do with conflicts of versions for the configuration library, night config. Currently, one of the things my mod relies on to check if the config file is valid isn't supported in the newest version of the lib, and when other mods bring in the lib it causes my mod to default to it and crash. It seems like the work to fix this [has been merged](https://github.com/TheElectronWill/night-config/commit/d474dc7620f08d148ea1c5b68690bcaa9c5beb95), but no release for it yet. I'll probably temporarily disable the check and release a beta version that should work until I release the full fix once night config releases a version with the merged fix.

**Comment by Quplet** [comment]
@mpustovoi @KAGUYA-1137 @sternschnaube @poism114514 @GanerCodes check releases for 1.0.1-beta, see if it fixes it.

**Comment by mpustovoi** [comment]
> 1.0.1-beta, see if it fixes it

I tested it on my repaired instance. Everything is working fine, there is no crash.

**Comment by poism114514** [comment]
> @mpustovoi @KAGUYA-1137 @sternschnaube @poism114514 @GanerCodes check releases for 1.0.1-beta, see if it fixes it.

I also tested and no crash again!

**Comment by venaminecraft** [comment]
Thanks for the update and for testing. No server crash anymore 😃

**Comment by mpustovoi** [comment]
You're not gonna believe me. Crash appeared again.
Although I probably just missed it, since when I entered my test world before and everything worked fine.
At the moment, when I click the `Create a New World` button, the game crashes immediately with some kind of `mouseClicked event handler` error.
I removed `1.0.1-beta` and everything is fine. I put it back and try again, crash.
Here are the [`logs`](https://mclo.gs/lNAGWlD).
I can't send my `Modrinth` instance here because now it has somehow become 60 MB in size.
